src/seedu/addressbook/data/person/Person.java

```diff
@@ -81,7 +81,30 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return getAsTextShowAll();
+        Printable[] printables = new Printable[4];
+        printables[0] = name;
+        if (!getPhone().isPrivate()) {
+        	printables[1] = getPhone();
+        }
+        if (!getEmail().isPrivate()) {
+        	printables[2] = getEmail();
+        }
+        if (!getAddress().isPrivate()) {
+        	printables[3] = getAddress();
+        }
+        return getPrintableString(printables);
+    }
+    
+    public String getPrintableString(Printable...printables){
+    	StringBuilder builder = new StringBuilder();
+    	for(Printable printable: printables){
+    		if (printable == null) {
+    			continue;
+    		}
+    		builder.append(printable.getPrintableString());
+    		builder.append(' ');
```

Comment:
This results in an extra space at the end. You could have used `StringJoiner` or `String.join` instead.